azurelinuxagent/common/protocol/wire.py

```diff
@@ -717,42 +716,35 @@ def update_certs(self, goal_state):
         xml_text = self.fetch_config(goal_state.certs_uri,
                                      self.get_header_for_cert())
         self.save_cache(local_file, xml_text)
-        self.certs = Certificates(self, xml_text)
+        self.set_certs(Certificates(self, xml_text))
 
     def update_remote_access_conf(self, goal_state):
         if goal_state.remote_access_uri is None:
             # Nothing in accounts data.  Just return, nothing to do.
             return
         xml_text = self.fetch_config(goal_state.remote_access_uri,
                                      self.get_header_for_cert())
-        self.remote_access = RemoteAccess(xml_text)
-        local_file = os.path.join(conf.get_lib_dir(), REMOTE_ACCESS_FILE_NAME.format(self.remote_access.incarnation))
+        remote_access = RemoteAccess(xml_text)
+        local_file = os.path.join(conf.get_lib_dir(), REMOTE_ACCESS_FILE_NAME.format(remote_access.incarnation))
         self.save_cache(local_file, xml_text)
+        self.set_remote_access(remote_access)
 
     def get_remote_access(self):
-        incarnation_file = os.path.join(conf.get_lib_dir(),
-                                        INCARNATION_FILE_NAME)
-        incarnation = self.fetch_cache(incarnation_file)
-        file_name = REMOTE_ACCESS_FILE_NAME.format(incarnation)
-        remote_access_file = os.path.join(conf.get_lib_dir(), file_name)
-        if not os.path.isfile(remote_access_file):
-            # no remote access data.
-            return None
-        xml_text = self.fetch_cache(remote_access_file)
-        remote_access = RemoteAccess(xml_text)
-        return remote_access
+        if self._remote_access is None:
+            raise ProtocolError("Trying to fetch Remote Access before initialization!")
```

Comment:
seems like the previous code was returning None rather than raising

Reply:
Yes, I was initially just setting it to None too but changed to raise a ProtocolError to keep it at par with the rest of the code. I am handling this now in the remoteaccess.py file
```python
                    try:
                        self.remote_access = self.protocol.client.get_remote_access()
                    except ProtocolError:
                        self.remote_access = None
```